src/app/login/login.component.ts

```diff
@@ -1,30 +1,55 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
 import { FormGroup, Validators } from '@angular/forms';
 import { FormBuilder } from '@angular/forms';
+import { Router } from '@angular/router';
+import { catchError } from 'rxjs';
+import { AccountService } from '../services/account.service';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.less']
 })
 export class LoginComponent implements OnInit {
- constructor(private fb : FormBuilder){}
- loginForm!: FormGroup;
 
+ constructor(private fb : FormBuilder,
+             private accountService: AccountService,
+             private router: Router){}
+ loginForm!: FormGroup;
+ errorMessage?:string;
   ngOnInit(): void {
-
-   this.loginForm = this.fb.group({
+    this.loginForm = this.fb.group({
     email: ['', [Validators.required, Validators.email]],
     password: ['', Validators.required]
    })
   }
-  
+  //getters for form access 
   get password(){return this.loginForm.get('password'); }
   get email(){ return this.loginForm.get('email'); }
 
-  login():void{
-    console.warn("login called");
+  submit(){
+    //return if form is not valid
+    if(this.loginForm.invalid){return;}
+   
+    this.accountService.login(this.email?.value, this.password?.value)  
+   .subscribe({
```

Comment:
Anything that is subscribed, must also complete that observable otherwise that observable will forever be waiting for another input. You can do this many ways. `takeUntil`, save a reference to the subscription and unsubscribe in an OnDestroy. In this case I recommend using `take(1)` in the `pipe` of the observable. Take 1 means you're only expecting one emit from the observable and it will complete after that.

These are critical topics to learn for advanced RXJS/Angular. If you can find somewhere to read up on this info it would be beneficial.